external-crates/move/crates/move-compiler/src/naming/translate.rs

```diff
@@ -62,10 +63,17 @@ struct ResolvedModuleFunction {
     ty_args: Option<Vec<N::Type>>,
 }
 
-#[derive(PartialEq)]
-enum NominalBlockType {
+#[derive(PartialEq, Eq, Copy, Clone, Debug)]
+enum LoopType {
+    While,
     Loop,
+}
+
+#[derive(PartialEq, Eq, Copy, Clone, Debug)]
+enum NominalBlockType {
+    Loop(LoopType),
     Block,
+    Lambda,
```

Comment:
I think we should split this up:

```suggestion
    LambdaBody,
    LambdaBreak,
```

Then a lambda can push two scopes. I think this will work out better long-term: if we decide `Lambda` does not capture break, we can remove the `LambdaBreak` scope; if we decide it does, we'll be able to decorate those labels onto the lambda definitions for expansion downstream.